### PR TITLE
Private events feature

### PR DESCRIPTION
**Pull request by Jarfsoft**
**Features added:**
- Create new profile
- Sign in
- Create events
- Attend event
- Edit and delete profile

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+         :recoverable, :rememberable, :validatable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  validates :password, length: { minimum: 6 }
+  # has_many :invitations
+  # has_many :events
+  has_many :events, foreign_key: 'creator_id', class_name: 'Event'
+  # has_many :edited_posts, foreign_key: :editor_id, class_name: "Post"
+  has_many :attended_events, through: :attendances, source: :event
+end
```
Comment (oloomoses) [inline app/models/user.rb]: Remember to delete all the commented code

**Review by oloomoses — CHANGES_REQUESTED** [review]
# Status: Changes required
Hello team, you are doing great on this project
kindly implement the changes below before submitting another review

- [x] check that there is a link to the event show page
- [x] Ensure that the event's show page displays the list of attendants 
- [x] Add unit tests for the models
- [x] Also remember to fix the logout. it is breaking

##### UsersController
###### `show` action
* [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

Feel free to reach me on slack [Oloo Moses]
Oloo Moses

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Request changes:full_moon:
**Hi team!
Great job on implementing the changes requested by the previous Tse :clap::clap:. However, we just need to make a few changes to get this project approved :muscle:.**
**Mandatory changes**

- [x] an index for the creator_id column is added to the events table. This is missing in the event table. Kindly fix this by creating and adding the migrations.

- [x] the events_attendation/events_users/(any other name for join table between users and events) table has column `user_id` of type `integer`. This is also missing. Kindly add it

- [x] an index for `user_id` column is added. Also, add this.

[Optional ]
  - Make the UI more user-friendly by styling it. 
  
Make the requested changes and when you are done, kindly request for another review.

You can reach me on slack @ Eric Enaburekhan if you have any questions.
Happy Coding:blush:
Eric Enaburekhan

**Review by layzrd — CHANGES_REQUESTED** [review]
### Status: Require changes 🌳 ♲
Hi @Jarfsoft, I am  [Abdulaziz Redi Ali](https://azizali.ml/), your TSE for this PR.  👏👏   I appreciate your hard work. Kindly fix the following issues. 

[REQUIRED]

- [ ] The project misses the edit.html file in the events view/folder.

Once you fix the issues, kindly submit another code review. ♲

Happy coding 👨‍💻
[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team  ,  you have done a brilliant job with this project. :clap:
#### You can merge !!!

**Pull request merged**